wifiphisher.py

```diff
@@ -881,32 +884,127 @@ def get_hostapd():
     # Get hostapd if needed
     get_hostapd()
 
+    # get all the local templates
+    local_templates = phishingpage.get_local_templates()
+
+    # loop through the local templates
+    for name in local_templates:
+
+        # add the template to the database
+        phishingpage.add_template(name)
+
     # get template_database
     template_database = phishingpage.get_template_database()
 
-    # check to see if the template is local
-    if not template_database[TEMPLATE_NAME] is None:
+    # get all the templtes names for display
+    template_names = list(template_database.keys())
+
+    # variable used to indicate correct template selection
+    template_selected = False
+
+    # loop until all operations for template selection is done
+    while True:
+
+        # check if the template argument is set and is correct
+        if args.template and args.template in template_database:
+
+            # set the template name
+            template_name = args.template
+
+            # set the template selected to avoid interactive selection
+            template_selected = True
+
+        # loop until a valid template is selected
+        while not template_selected:
 
-        # if template is incomplete locally, delete and ask for a download
-        if not phishingpage.check_template(TEMPLATE_NAME):
+            # display start of template names
+            print "\nList Of All Templates: \n"
 
-            # clean up the previous download
-            phishingpage.clean_template(TEMPLATE_NAME)
+            # display the templates
+            for number in range(len(template_names)):
+                print ("[" + G + str(number + 1) + W + "] " +
+                       template_names[number])
+
+            # get user's choice
+            choosen_template = raw_input("Please Select One Of The Above "
+                                         "Templates: ")
+
+            # try to convert the input to integer
+            try:
+
```

Comment:
There are a lot of whitespaces everywhere. Please, try to stick to the coding style.